Badly rewritten URLs in case of "/" in the article title

The forward slash in the URL is causing problems in Kiwix JS (see https://github.com/kiwix/kiwix-js/issues/494). I wonder if it might be causing problems for you too. For example, the article `Singapore` references its assets like this:
```
<link href="../-/s/css_modules/ext.kartographer.link.css" …>
```
Whereas the article `Singapore/Riverside` references its assets like this:
```
<link href="../../-/s/css_modules/ext.kartographer.link.css" …>
```
Note the extra `../../`, because the browser thinks it's accessing an article Riverside in the subdirectory Singapore. This would all be fine, except that the hrefs in the `Singapore/Riverside` article are written like this:
```
<p>The <b>Singapore River</b> forms a central artery in <a href="Singapore"
title="Singapore">Singapore</a>'s densely packed Central Business District.
```
Note that to access the hyperlink on this page from a browser, we would need to write that link as `<a href="../Singapore" ...>`.

So there seems to be some inconsistency. Are we inside a subdirectory from the browser's perspective, or not? The subroutine that writes the location of the assets seems to think so, while the one that writes the location of the hyperlinks doesn't! Do you think this is an mwoffliner issue?

No, it's not, I have created a new ZIM file with git master head of wikinews FR, and here is what `zimcheck` reports (we have broken internal links):
```
$ zimcheck out/wikinews_fr_all_2019-07.zim 
[INFO] Checking zim file out/wikinews_fr_all_2019-07.zim
[INFO] Verifying Internal Checksum.. 
  [INFO] Internal checksum found correct
[INFO] Searching for metadata entries..
[INFO] Searching for Favicon..
[INFO] Searching for main page..
[INFO] Verifying Articles' content.. 
[INFO] Searching for redundant articles..
  Verifying Similar Articles for redundancies..
[ERROR] Invalid internal links found :
  A/Belgique_:_Belgacom/Évènements_du_10_août_2014 (%C3%89v%C3%A8nements_du_10_ao%C3%BBt_2014) was not found in article A/Belgique_:_Belgacom/swing.be,_le_médiateur_des_télécoms_s'interroge_et_déplore_l'attitude_de_Belgacom.
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_1_:_Les_résultats_de_la_vingt-septième_journée (Championnat_de_France_2007%2F2008_de_Ligue_1_%3A_Les_r%C3%A9sultats_de_la_vingt-septi%C3%A8me_journ%C3%A9e#cite_ref-1) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_1_:_Les_résultats_de_la_vingt-septième_journée
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_1_:_Lyon_vainqueur_à_Toulouse_et_Lille_à_Marseille (Championnat_de_France_2007%2F2008_de_Ligue_1_%3A_Lyon_vainqueur_%C3%A0_Toulouse_et_Lille_%C3%A0_Marseille#cite_ref-2) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_1_:_Lyon_vainqueur_à_Toulouse_et_Lille_à_Marseille
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_1_:_le_suspense_continue (Championnat_de_France_2007%2F2008_de_Ligue_1_%3A_le_suspense_continue#cite_ref-euro_2-0) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_1_:_le_suspense_continue
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_1_:_les_résultats_de_la_12ème_journée (Championnat_de_France_2007%2F2008_de_Ligue_1_%3A_les_r%C3%A9sultats_de_la_12%C3%A8me_journ%C3%A9e#cite_ref-1) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_1_:_les_résultats_de_la_12ème_journée
  A/Championnat_de_France_2007/Football_:_décès_d'Antonio_Puerta (Football_%3A_d%C3%A9c%C3%A8s_d'Antonio_Puerta) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_1_:_les_résultats_de_la_6ème_journée_(suite)
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_1_:_les_résultats_de_la_trente-cinquième_journée (Championnat_de_France_2007%2F2008_de_Ligue_1_%3A_les_r%C3%A9sultats_de_la_trente-cinqui%C3%A8me_journ%C3%A9e#cite_ref-2) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_1_:_les_résultats_de_la_trente-cinquième_journée
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_1_:_les_résultats_de_la_trente-quatrième_journée (Championnat_de_France_2007%2F2008_de_Ligue_1_%3A_les_r%C3%A9sultats_de_la_trente-quatri%C3%A8me_journ%C3%A9e#cite_ref-2) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_1_:_les_résultats_de_la_trente-quatrième_journée
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_1_:_les_résultats_de_la_trente-troisième_journée (Championnat_de_France_2007%2F2008_de_Ligue_1_%3A_les_r%C3%A9sultats_de_la_trente-troisi%C3%A8me_journ%C3%A9e#cite_ref-2) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_1_:_les_résultats_de_la_trente-troisième_journée
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_2_:_Le_Havre_AC_en_Ligue_1_la_saison_prochaine (Championnat_de_France_2007%2F2008_de_Ligue_2_%3A_Le_Havre_AC_en_Ligue_1_la_saison_prochaine#cite_ref-l1_1-0) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_2_:_Le_Havre_AC_en_Ligue_1_la_saison_prochaine
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_2_:_Match_nul_entre_Sedan_et_Bastia (Championnat_de_France_2007%2F2008_de_Ligue_2_%3A_Match_nul_entre_Sedan_et_Bastia#cite_ref-nat_2-0) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_2_:_Match_nul_entre_Sedan_et_Bastia
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_2_:_Une_saison_se_termine (Championnat_de_France_2007%2F2008_de_Ligue_2_%3A_Une_saison_se_termine#cite_ref-1) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_2_:_Une_saison_se_termine
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_2_:_le_FC_Nantes_en_Ligue_1_la_saison_prochaine (Championnat_de_France_2007%2F2008_de_Ligue_2_%3A_le_FC_Nantes_en_Ligue_1_la_saison_prochaine#cite_ref-nat_2-0) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_2_:_le_FC_Nantes_en_Ligue_1_la_saison_prochaine
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_2_:_les_résultats_de_la_trente-quatrième_journée (Championnat_de_France_2007%2F2008_de_Ligue_2_%3A_les_r%C3%A9sultats_de_la_trente-quatri%C3%A8me_journ%C3%A9e#cite_ref-nat_2-0) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_2_:_les_résultats_de_la_trente-quatrième_journée
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_2_:_match_nul_entre_Angers_et_Nantes (Championnat_de_France_2007%2F2008_de_Ligue_2_%3A_match_nul_entre_Angers_et_Nantes#cite_ref-nat_2-0) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_2_:_match_nul_entre_Angers_et_Nantes
  A/Championnat_de_France_2007/Championnat_de_France_2007/2008_de_Ligue_2_:_score_vierge_entre_Grenoble_et_Ajaccio (Championnat_de_France_2007%2F2008_de_Ligue_2_%3A_score_vierge_entre_Grenoble_et_Ajaccio#cite_ref-1) was not found in article A/Championnat_de_France_2007/2008_de_Ligue_2_:_score_vierge_entre_Grenoble_et_Ajaccio
  I/m/Flag_of_Scotland_(traditional).svg.png (../I/m/Flag_of_Scotland_(traditional).svg.png) was not found in article A/Championnat_de_Ligue_celtique_2014-2015_:_les_résultats_de_la_dix-neuvième_journée
  A/Congo/Évènements_du_23_avril_2019 (%C3%89v%C3%A8nements_du_23_avril_2019) was not found in article A/Congo/Rwanda_:_une_quarantaine_de_corps_retrouvés_après_naufrage_sur_lac_Kivu
  A/De_l'alcool_et_du_sucre_dans_la_comète_C/Évènements_du_1er_novembre_2015 (%C3%89v%C3%A8nements_du_1er_novembre_2015) was not found in article A/De_l'alcool_et_du_sucre_dans_la_comète_C/2014_Q2_(Lovejoy)
  A/Découverte_d'oxygène_sur_la_comète_67P/Évènements_du_31_octobre_2015 (%C3%89v%C3%A8nements_du_31_octobre_2015) was not found in article A/Découverte_d'oxygène_sur_la_comète_67P/Tchourioumov-Guérassimenko
  A/Europe_de_l'Ouest_:_vigilance_orange_neige/Évènements_du_9_février_2013 (%C3%89v%C3%A8nements_du_9_f%C3%A9vrier_2013) was not found in article A/Europe_de_l'Ouest_:_vigilance_orange_neige/verglas_en_France,_en_Belgique_et_au_Luxembourg
  A/France/Évènements_du_15_avril_2017 (%C3%89v%C3%A8nements_du_15_avril_2017) was not found in article A/France/Suisse_:_mort_de_l'auteur_des_tirs_de_1982_contre_la_centrale_de_Creys-Malville
  A/Les_États-Unis_font_usage_d'une_GBU-43/Évènements_du_12_avril_2017 (%C3%89v%C3%A8nements_du_12_avril_2017) was not found in article A/Les_États-Unis_font_usage_d'une_GBU-43/B_pour_la_première_fois_en_Afghanistan
  A/Passage_de_la_comète_C/Évènements_du_12_mars_2013 (%C3%89v%C3%A8nements_du_12_mars_2013) was not found in article A/Passage_de_la_comète_C/2011_L4_(PANSTARRS)
[ERROR] Invalid external links found :
  https://commons.wikimedia.org/w/api.php?action=timedtext&amp;title=File%3APresident_Obama_on_Death_of_Osama_bin_Laden.ogv&amp;lang=cs&amp;trackformat=srt&amp;origin=%2Ais an external dependence in article A/2011_sur_Wikinews
  https://commons.wikimedia.org/w/api.php?action=timedtext&amp;title=File%3ABangladesh_building_collapse_-_WN.ogv&amp;lang=bn&amp;trackformat=srt&amp;origin=%2Ais an external dependence in article A/Bangladesh_:_effondrement_mortel_d'un_immeuble
  https://commons.wikimedia.org/w/api.php?action=timedtext&amp;title=File%3ABangladesh_building_collapse_-_WN.ogv&amp;lang=bn&amp;trackformat=srt&amp;origin=%2Ais an external dependence in article A/Bangladesh_:_émeutes_et_revendications_après_l'effondrement_mortel_d'un_immeuble
  https://commons.wikimedia.org/w/api.php?action=timedtext&amp;title=File%3AWhite_House_Spokesman_Spicer_Holds_News_Conference.webm&amp;lang=en&amp;trackformat=srt&amp;origin=%2Ais an external dependence in article A/Le_porte-parole_de_la_Maison_Blanche_attaque_les_médias_lors_de_sa_première_allocution
  https://commons.wikimedia.org/w/api.php?action=timedtext&amp;title=File%3APresident_Obama_on_Death_of_Osama_bin_Laden.ogv&amp;lang=cs&amp;trackformat=srt&amp;origin=%2Ais an external dependence in article A/Oussama_ben_Laden_tué_lors_d'un_raid_près_d'Islamabad
  https://commons.wikimedia.org/w/api.php?action=timedtext&amp;title=File%3AWalesCalltoAction.ogv&amp;lang=ca&amp;trackformat=srt&amp;origin=%2Ais an external dependence in article A/Wikipédia_célèbre_son_dixième_anniversaire_en_ligne
  https://commons.wikimedia.org/w/api.php?action=timedtext&amp;title=File%3APresident_Obama_speaks_on_attacks_in_Boston_%282013-04-16%29.ogv&amp;lang=en&amp;trackformat=srt&amp;origin=%2Ais an external dependence in article A/États-Unis_:_3_morts_et_plus_de_170_blessés_dans_l'attentat_de_Boston
  https://commons.wikimedia.org/w/api.php?action=timedtext&amp;title=File%3APresident_Obama_on_Death_of_Osama_bin_Laden.ogv&amp;lang=cs&amp;trackformat=srt&amp;origin=%2Ais an external dependence in article A/États-Unis_:_Obama_demande_une_enquête_sur_Ben_Laden
  https://commons.wikimedia.org/w/api.php?action=timedtext&amp;title=File%3APresident_Obama_speaks_on_explosions_in_Boston_%282013-04-15%29.ogv&amp;lang=en&amp;trackformat=srt&amp;origin=%2Ais an external dependence in article A/États-Unis_:_attentats_à_Boston
[INFO] Overall Test Status: Fail
[INFO] Total time taken by zimcheck: 272 seconds.
```

--- a/src/Dump.ts
+++ b/src/Dump.ts
@@ -218,7 +218,7 @@ export class Dump {
     }
 
     public getArticleUrl(articleId: string) {
-        return this.getArticleBase(articleId, true);
+        return this.getArticleBase(articleId, false);
     }
 
     public getArticleBase(articleId: string, escape?: boolean) {

--- a/test/unit/urlRewriting.test.ts
+++ b/test/unit/urlRewriting.test.ts
@@ -71,7 +71,7 @@ test('Url re-writing', async (t) => {
 
     await rewriteUrl(complexParentArticleId, mw, dump, $wikiLinkWithSlash);
     t.assert($wikiLinkWithSlash.nodeName === 'A', 'wikiLinkWithSlash is still a link');
-    t.equal($wikiLinkWithSlash.getAttribute('href'), '../../A/Farnborough%2FAldershot_Built-up_Area', 'wikiLinkWithSlash HREF is correct');
+    t.equal($wikiLinkWithSlash.getAttribute('href'), '../../A/Farnborough/Aldershot_Built-up_Area', 'wikiLinkWithSlash HREF is correct');
 
     await rewriteUrl(complexParentArticleId, mw, dump, $specialMap);
     t.assert($specialMap.nodeName === 'A', 'specialMap is still a link');

